src/phoenix/server/api/types/PromptVersionTemplate.py

```diff
@@ -0,0 +1,89 @@
+# Part of the Phoenix PromptHub feature set
+
+from enum import Enum
+from typing import Any, Union
+
+import strawberry
+from pydantic import BaseModel
+from strawberry.scalars import JSON
+
+JSONSerializable = Union[None, bool, int, float, str, dict[str, Any], list[Any]]
+
+
+@strawberry.enum
+class PromptMessageRole(str, Enum):
+    USER = "user"
+    SYSTEM = "system"
+    AI = "ai"  # E.g. the assistant. Normalize to AI for consistency.
+
+
+class PromptStringTemplate(BaseModel):
+    template: str
+
+
+class TextPromptMessage(BaseModel):
+    role: PromptMessageRole
+    content: str
+
+    def to_gql(self):
+        return TextPromptMessageGQL.from_model(self)
+
+
+class JSONPromptMessage(BaseModel):
+    role: PromptMessageRole
+    content: JSONSerializable
+
+    def to_gql(self):
+        return JSONPromptMessageGQL.from_model(self)
+
+
+class PromptMessagesTemplateV1(BaseModel):
+    _version: str = "messages-v1"
+    template: list[Union[TextPromptMessage, JSONPromptMessage]]
```

Comment:
`message_templates` or `messages`?

Reply:
`ChatTemplateV1`?

Reply:
I think it's probably a good idea for the different template types to conform to the same interface for now

Reply:
This should just be a union no?

Reply:
ah right

Reply:
wait no, I think it's list[Union[...]] since there can be multiple messages

Reply:
they key should be `messages` that's what's confusing here

Reply:
ah I see, I changed it to `template` so all of our template wrappers have the same interface which I'm kind of partial to (the wrapper for the plain string template just has a `template` key)

Reply:
It shouldn't leak into the graphQL type